### PR TITLE
Kaggle

### PR DESCRIPTION
**Pull request by nnistelrooij**
Add inference loop with last 365 days of available data.

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -35,20 +35,25 @@ def __init__(self, num_const, num_var, num_hidden, num_out, num_groups, num_subm
...
                                                  num_var, 
-                                                 num_hidden, 
-                                                 num_out, 
+                                                 num_hidden,
+                                                 num_out,
                                                  num_groups,
-                                                 independent)
-                                        for num_groups in self.num_groups])
+                                                independent)
+                                        for num_groups in self.num_groups]
```
Comment (nnistelrooij) [inline model.py]: Missing closing parenthesis

File: utils/data.py [hunk 2]
```diff
@@ -207,8 +204,8 @@ def __getitem__(self, idx):
                 where 1 <= |targets| <= horizon
 
         If horizon = 0, i.e. inference mode, then sales may be empty, which
-        means that it needs to be returned separately. So then day, items
-        without sales, and sales separately are returned.
+        means that items may not contian sales. So then day and items
```
Comment (rusane) [inline utils/data.py:207]: Minor typo: contian -> contain

**Review by rusane — APPROVED** [review]
Looks good to me!

File: model.py [hunk 3]
```diff
@@ -35,20 +35,25 @@ def __init__(self, num_const, num_var, num_hidden, num_out, num_groups, num_subm
...
+        self.device = device
+                                       
     def reset_hidden(self):
         for submodel in submodels:
-            submodel.reset_hidden()
+            submodel.reset_hidden()     
 
-    def forward(self, items, day):
+    def forward(self, day, items):
+        day = day.to(self.device)
```
Comment (MauiGonzo) [inline model.py:51]: at this point an error occurs:
  File "/Users/mauriceverbrugge/github/MLiP-2020/model.py", line 55, in forward
    for i, items in enumerate(items.split(self.num_groups, dim=-1)):
  File "/usr/local/lib/python3.7/site-packages/torch/tensor.py", line 386, in split
    return super(Tensor, self).split_with_sizes(split_size, dim)

File: model.py [hunk 4]
```diff
@@ -19,14 +19,14 @@ def __init__(self, num_const, num_var, num_hidden, num_out, num_groups, independ
...
-        lstm_out = self.bn(lstm_out[:, -1]) # take last day from sequence
+    def forward(self, day, items):
+        lstm_out, hidden = self.lstm(items, day)
+        lstm_out = lstm_out[:, -1] # take last day from sequence
         y = self.fc(lstm_out)
         return y
 
 class Model(nn.Module):
-    def __init__(self, num_const, num_var, num_hidden, num_out, num_groups, num_submodels, independent=True):
+    def __init__(self, num_const, num_var, num_hidden, num_out, num_groups, num_submodels, device, independent=True):
```
Comment (MauiGonzo) [inline model.py:29]: requires change in call in main routine in this file

**Review by MauiGonzo — COMMENTED** [review]
in quick check run time-error encountered

**Comment by nnistelrooij** [comment]
I changed some dimensions around. It should work in the `clean` branch. I had to do that to make sure the sub-layers stayed independent of each other.
